### PR TITLE
Import `ultralytics.nn.tasks` as `ultralytics.nn`

### PR DESCRIPTION
**Pull request by AyushExel**
I'm working on using ultralytics package for some of the yolov5 operations and I realized that we should support better import structures that feel natural and easy to remember.
Before:
```
from ultralytics.nn.tasks import attempt_load_weights
...
```
Now
```
from ultralytics.nn import attempt_load_weights
```

### Summary
📦🧹🔧

This pull request simplifies the import statements and improves the batch normalization fusion in the `ultralytics.nn` package. It also makes the `tasks.py` module more accessible by importing its contents into the package's `__init__.py` module.

> _We are the masters of the code, we simplify and fuse_
> _We import from the `nn` package, we have nothing to lose_
> _We make the models run faster, we optimize the flow_
> _We are the ultralytics, we rock the machine learning show_

### Walkthrough
*  Simplify the import of tasks-related functions and classes from `ultralytics.nn` by exposing them in the `__init__.py` module ([link](https://github.com/ultralytics/ultralytics/pull/2889/files?diff=unified&w=0#diff-0d7515eac7ded4296ccd3f7e51ad31e4a8ba1320f078c5c76fc403fb96d926c2L1-R2), [link](https://github.com/ultralytics/ultralytics/pull/2889/files?diff=unified&w=0#diff-e2650ac65b6525f6d9c0594a226066d396afbcba637c054d29abe04e62f410c4L11-R13))
*  Remove the `Conv2` class from the batch normalization fusion check in the `fuse` method of the `BaseModel` class, since it already handles the fusion internally ([link](https://github.com/ultralytics/ultralytics/pull/2889/files?diff=unified&w=0#diff-e2650ac65b6525f6d9c0594a226066d396afbcba637c054d29abe04e62f410c4L105-R105))

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhancement of the neural network module with model utility functions and base classes.

### 📊 Key Changes
- Added import statements for base and specialized model classes (`BaseModel`, `ClassificationModel`, `DetectionModel`, `SegmentationModel`).
- Included utility functions for model weight loading and parsing (`attempt_load_one_weight`, `attempt_load_weights`, `parse_model`, etc.).
- Made these classes and functions accessible from the `ultralytics.nn` package by updating the `__all__` list.

### 🎯 Purpose & Impact
- 👍 **Purpose:** Streamlines the process of loading, parsing, and identifying different types of models, making it easier for developers to work with various machine learning tasks.
- 💡 **Impact:** Provides users with more tools out-of-the-box, potentially improving user experience when interfacing with Ultralytics AI models. Developers can now load and infer model details more efficiently, improving productivity.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@AyushExel this looks good!

But make sure you update your local code, because your recent PRs keep trying to delete the Conv2() fuse lines that are in main that are unrelated to your PRs:

Screenshot 2023-05-29 at 13 04 10

**Comment by AyushExel** [comment]
@glenn-jocher It shows I'm up to date with main
Screenshot 2023-05-29 at 4 37 16 PM

**Comment by AyushExel** [comment]
Okay should be good now

**Comment by glenn-jocher** [comment]
@AyushExel well something seems wrong, because the lines are here in main:
https://github.com/ultralytics/ultralytics/blob/a2bb42dfe96e39cd1513740236424b8ad61b5270/ultralytics/nn/tasks.py#L105-L107

We want to keep these, but it seems your local code is missing them so each PR tries to remove it.

**Comment by glenn-jocher** [comment]
Conv2 blocks are parallel convolutions that get fused into 1 convolution with the .fuse() op. I created these based on YOLOv6 RepConv block.

**Comment by glenn-jocher** [comment]
@AyushExel looks good!

**Comment by glenn-jocher** [comment]
@AyushExel PR merged!
